src/cephadm/cephadm

```diff
@@ -108,13 +108,25 @@ def port_in_use(port_num):
     # type (int) -> bool
     """Detect whether a port is in use on the local machine - IPv4 and IPv6"""
 
+    def attempt_bind(address):
+        # type (str) -> None
+        try:
+            s.bind((address, port_num))
+        except (socket.error, OSError) as e:  # py2 and py3
+            if e.errno == 98:                 # Address already in use
+                raise OSError
+            elif e.errno == 99:               # Cannot assign requested address
+                return
+
     try:
         s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-        s.bind(("127.0.0.1", port_num))
```

Comment:
Add arg to specify the ip addr during deploy? 

e.g. `cephadm deploy --mon-ip 0.0.0.0` or similar?

Reply:
possibly - but I think on a deploy we should validate the IP exists on the host AND the port(s) are free - which is bigger than the scope I was intending for this PR